src/web/bridges/hooks/use-bridge.ts

```diff
@@ -0,0 +1,51 @@
+import type { McpUiHostContext } from "@modelcontextprotocol/ext-apps";
+import { useSyncExternalStore } from "react";
+import { AppsSdkBridge } from "../apps-sdk-bridge";
+import { McpAppBridge } from "../mcp-app-bridge";
+
+export type BridgeInterface = Required<
+  Pick<McpUiHostContext, "theme" | "locale">
+>;
+
+type BridgeExternalStore<K extends keyof BridgeInterface> = {
+  subscribe: (onChange: () => void) => () => void;
+  getSnapshot: () => BridgeInterface[K];
+};
+
+const DEFAULT_VALUE_FOR_MCP_APP_BRIDGE: BridgeInterface = {
+  theme: "light",
+  locale: "en-US",
+};
+
+const getExternalStore = <K extends keyof BridgeInterface>(
+  key: K,
+  defaultValue: BridgeInterface[K] = DEFAULT_VALUE_FOR_MCP_APP_BRIDGE[key],
+): BridgeExternalStore<K> => {
+  const hostType = window.skybridge.hostType;
+  if (hostType === "apps-sdk") {
+    const bridge = AppsSdkBridge.getInstance();
+    return {
+      subscribe: bridge.subscribe(key),
+      getSnapshot: () => bridge.getSnapshot(key) as BridgeInterface[K],
+    };
+  }
+  const bridge = McpAppBridge.getInstance();
+  return {
+    subscribe: bridge.subscribe(key),
+    getSnapshot: () =>
+      (bridge.getSnapshot(key) ?? defaultValue) as BridgeInterface[K],
```

Comment:
Casting here

Reply:
You may be able not to cast if getExternalStore does not take key as an argument and return uncalled bridge methods, don't you think?